vnext/include/ReactUWP/Views/KeyboardEventHandler.h

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include <optional>
+#include <set>
+#include <folly/dynamic.h>
+#include <winrt/Windows.UI.Xaml.Controls.h>
+#include <winrt/Windows.UI.Xaml.Input.h>
+
+#include "XamlView.h"
+#include <IReactInstance.h>
+
+namespace winrt {
+  using namespace Windows::UI;
+  using namespace Windows::UI::Xaml;
+  using namespace Windows::UI::Xaml::Controls;
+  using namespace Windows::UI::Xaml::Input;
+  using namespace Windows::Foundation;
+  using namespace Windows::UI::Xaml::Media;
+  using namespace Windows::System;
+  using namespace Windows::UI::Core;
+}
+
+using namespace std;
+
+namespace react { namespace uwp {
+  enum class HandledEventPhase
+  {
+    CAPTURING = 1, // match the value with EventPhase in React. EventPhase includes None, Capturing, AtTarget, Bubbling
+    BUBBLING = 3
+  };
+
+  struct ModifiedKeyStatue
+  {
+    bool altKey{ false };
+    bool ctrlKey{ false };
+    bool metaKey{ false };
+    bool shiftKey{ false };
+    bool capLocked{ false };
+  };
+
+  struct ReactKeyboardEvent: ModifiedKeyStatue
+  {
+    int64_t target = 0;
+    string key{};
+  };
+
+  struct KeyboardEvent: ModifiedKeyStatue
+  {
+    HandledEventPhase handledEventPhase{ HandledEventPhase::BUBBLING };
+    string key{};
+  };
+
+  typedef std::function<void(winrt::IInspectable const&, winrt::KeyRoutedEventArgs const&)> KeyboardEventCallback;
```

Comment:
You're using the std namespace but inconsistent throughout the file about including the prefix or not. Scrub the file for that or remove the using statement.

Reply:
Since this is a header, the preference would be to remove it to not pollute other files. I realize std:: is low risk here.

---
In reply to: [295559925](https://github.com/microsoft/react-native-windows/pull/2623#discussion_r295559925) [](ancestors = 295559925)

Reply:
I'm attempting to tread a little light on style here since I arrived here... /checks watch... moments ago. But in general I'd consider bringing a namespace (even std) into a header to be a forbidden act. It can create all sorts of havok.

Reply:
And then there's the fact that multiple teams have been in this codebase using differing standards. And it has also included sometimes following the facebook::react patterns and sometimes not. As Keith called out in another comment, some work should be done to agree on basic standards. I want to say folks should follow what's already in the files until then, but that's hard because you can find so many different patterns.

---
In reply to: [295567659](https://github.com/microsoft/react-native-windows/pull/2623#discussion_r295567659) [](ancestors = 295567659)

Reply:
For std::vector, std::function, I'd like to keep std namespace. 
for string, I prefer not to prefix with std::

Reply:
Reconsider. Your preferences are not universal.

Reply:
This conversation is marked as closed but I consider this a pretty important issue. Using namespaces in a header can be very problematic. In the large this isn't done in the codebase (looks like FollyDynamicConverter.h is the only other offender). Please remove the using statement.